src/realm/object-store/schema.cpp

```diff
@@ -280,9 +285,15 @@ static void compare(ObjectSchema const& existing_schema, ObjectSchema const& tar
 }
 
 template <typename T, typename U, typename Func>
-void Schema::zip_matching(T&& a, U&& b, Func&& func) noexcept
+std::vector<ObjectSchema> Schema::zip_matching(T&& a, U&& b, Func&& func, bool is_schema_additive) noexcept
```

Comment:
This function is no longer `noexcept` as it now allocates memory.

It's also just no longer a generic `zip()` function but instead is something which only works on ObjectSchema. That's fine, but it means that there's no reason for it to be templated.

Reply:
This is going to make the code a little bit uglier. Since we were using ` &&`  for automatically binding/generating 2 versions of `zip_matching`. One with `const Schema& a, const Schema& b` for comparison and one for `Schema& a, const Schema& b` for updating or assigning a new property.
But I see your point, we don't need templates.